### DISCUSSION TITLE
Unit test failed

### DISCUSSION THREAD

**Discussion by QuinnVV** [Cyfrin/foundry-full-course-cu#1251] [Q&A] [answered]
I  run forge test (FundMeTest.t.sol that have 7 unit tests, 6 of them passed  and the other one (testPriceFeedSetCorrectly() got this error. Please help me on this

[FAIL. Reason: EvmError: Revert] testPriceFeedSetCorrectly() (gas: 12573)
Traces:
  [2985384] FundMeTest::setUp()
    ├─ [1485451] → new DeployFundMe@0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f
    │   └─ ← 7309 bytes of code
    ├─ [1443182] DeployFundMe::run()
    │   ├─ [890063] → new HelperConfig@0x104fBc016F4bb334D775a19E8A6510109AC63E00
    │   │   ├─ [0] VM::startBroadcast()
    │   │   │   └─ ← ()
    │   │   ├─ [377261] → new MockV3Aggregator@0x90193C961A926261B756D1E5bb255e67ff9498A1
    │   │   │   └─ ← 1107 bytes of code
    │   │   ├─ [0] VM::stopBroadcast()
    │   │   │   └─ ← ()
    │   │   └─ ← 2161 bytes of code
    │   ├─ [381] HelperConfig::activeNetworkConfig() [staticcall]
    │   │   └─ ← MockV3Aggregator: [0x90193C961A926261B756D1E5bb255e67ff9498A1]
    │   ├─ [0] VM::startBroadcast()
    │   │   └─ ← ()
    │   ├─ [486134] → new FundMe@0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496
    │   │   └─ ← 2316 bytes of code
    │   ├─ [0] VM::stopBroadcast()
    │   │   └─ ← ()
    │   └─ ← FundMe: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496]
    ├─ [0] VM::deal(0x0000000000000000000000000000000000000001, 10000000000000000000 [1e19])
    │   └─ ← ()
    └─ ← ()

  [12573] FundMeTest::testPriceFeedSetCorrectly()
    ├─ [2353] FundMe::getPriceFeed() [staticcall]
    │   └─ ← MockV3Aggregator: [0x90193C961A926261B756D1E5bb255e67ff9498A1]
    ├─ [0] 0x0000000000000000000000000000000000000000::activeNetworkConfig() [staticcall]
    │   └─ ← ()
    └─ ← EvmError: Revert
    here is my code
    
    `function testPriceFeedSetCorrectly() public {
        address retreivedPriceFeed = address(fundMe.getPriceFeed());
        // (address expectedPriceFeed) = helperConfig.activeNetworkConfig();
        address expectedPriceFeed = helperConfig.activeNetworkConfig();
        assertEq(retreivedPriceFeed, expectedPriceFeed);
    }`

**Comment by ghost** [comment]
This staticcall:

```
0x0000000000000000000000000000000000000000::activeNetworkConfig()
```

Makes me think you've not initialized `helperConfig` in the test file because it's trying to call the method on the `address(0)`.

Please, show me:

- Your test file state variables
- The `setUp` function

**Reply by QuinnVV** [reply]
Hi @pacelliv, here they are. Please take a look.

```
contract FundMeTest is Test {
    FundMe public fundMe;
    HelperConfig public helperConfig;

    uint256 public constant SEND_VALUE = 0.1 ether; // just a value to make sure we are sending enough!
    uint256 public constant STARTING_USER_BALANCE = 10 ether;
    uint256 public constant GAS_PRICE = 1;

    address public constant USER = address(1);

    function setUp() external {
        DeployFundMe deployer = new DeployFundMe();
        fundMe = deployer.run();

        vm.deal(USER, STARTING_USER_BALANCE);
    }
```

**Reply by ghost** [reply, marked as answer]
You haven't used the `helperConfig` returned by the `run` function. That's why the `helperConfig` in the test file defaults to the `address(0)`.

Update to:

```javascript
(fundMe, helperConfig) = deployer.run();
```

If this doesn't work, you need to check your deployment script and make sure it is returning an instance of the `HelperConfig` contract.
